### PR TITLE
Support user-provided instance connectors for subentity detail tabs

### PR DESCRIPTION
**Pull request by orangejenny**
Followup for https://github.com/dimagi/commcare-hq/pull/8785, specifically comment 1 here: https://github.com/dimagi/commcare-hq/pull/8785#issuecomment-150639065

@NoahCarnahan / @snopoke 

fyi @ctsims 

screen shot 2015-10-29 at 2 43 41 pm
screen shot 2015-10-29 at 2 43 47 pm

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/sections/entries.py [hunk 1]
```diff
@@ -190,6 +192,22 @@ def entry_for_module(self, module):
 
         return results
 
+
+    def _add_extra_entry_connectors(self, entry, module):
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/sections/entries.py]: E303 too many blank lines (2)

File: corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js [hunk 2]
```diff
@@ -532,6 +533,14 @@ var DetailScreenConfig = (function () {
                 that.header.setVisibleValue(visibleVal);
 
                 that.nodeset = uiElement.input().val(that.original.nodeset);
+                var o = {
+                    lang: that.lang,
+                    langs: that.screen.langs,
+                    items: that.original['connectors'],
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/detail-screen-config.js]: ['connectors'] is better written in dot notation.

**Comment by snopoke** [comment]
Is it not possible to figure out what is needed automatically like we do with the other instance declarations?

File: corehq/apps/app_manager/models.py [hunk 3]
```diff
@@ -1534,6 +1534,10 @@ class DetailTab(IndexedSchema):
     has_nodeset = BooleanProperty(default=False)
     nodeset = StringProperty()
 
+    # Any instance connectors necessary for the nodeset,
+    # e.g., "reports" => "jr://fixture/reports"
+    connectors = SchemaListProperty(MappingItem)
```
Comment (snopoke) [inline corehq/apps/app_manager/models.py]: Why can't this just be a `DictProperty` if you don't need the localization?

**Comment by snopoke** [comment]
I see why you can't auto-detect them because the app doesn't have any knowledge of them, poo.

File: corehq/apps/app_manager/models.py (hunk 3 above)
Comment (orangejenny) [inline corehq/apps/app_manager/models.py]: Yeah, that'd make this model more logical. Fixed.

**Pull request merged**